counterpartylib/lib/messages/broadcast.py

```diff
@@ -104,6 +104,19 @@ def validate (db, source, timestamp, value, fee_fraction_int, text, block_index)
         if len(text) > 52:
             problems.append('text too long')
 
+    if util.enabled('options_require_memo') and text and text.lower().startswith('options'):
+        ops_spl = text.split(" ")
+        if len(ops_spl) == 2:
+            try:
+                options_int = int(ops_spl.pop())
+
+                if (options_int > config.MAX_INT) or (options_int < 0):
+                    problems.append('integer overflow')
+                elif options_int > config.ADDRESS_OPTION_MAX_VALUE:
+                    problems.append('options out of range')
```

Comment:
needs test, at least for `options_int > config.ADDRESS_OPTION_MAX_VALUE`, preferably also the `MAX_INT` and `< 0`.

Reply:
also needs test for non INT value